### ISSUE TITLE
[Question] Batch the training

### ISSUE THREAD

**Issue by laifi** [facebookresearch/PyTorch-BigGraph#60]
Hi, i'm new to PBG , so i don't know about all the features and the functionalities .
i have just processed my edges file with this command :
```
torchbiggraph_import_from_tsv \
  --lhs-col=0 --rel-col=1 --rhs-col=2 \
  PyTorch-BigGraph/torchbiggraph/examples/configs/fb15k_config.py \
  edges.txt
```
**Output:**

> Looking up relation types in the edge files...
> - Found 727 relation types
> - Removing the ones with fewer than 1 occurrences...
> - Left with 727 relation types
> - Shuffling them...
> Searching for the entities in the edge files...
> Entity type all:
> - Found 3664382 entities
> - Removing the ones with fewer than 1 occurrences...
> - Left with 3664382 entities
> - Shuffling them...
> Preparing entity path data/FB15k:
> - Writing count of entity type all and partition 0
> - Writing count of dynamic relations
> ...

Then , i start the training : 
`torchbiggraph_train \
   PyTorch-BigGraph/torchbiggraph/examples/configs/fb15k_config.py \
  -p edge_paths=edges_partitioned`

**Output :** 

> 2019-06-11 09:55:28  Loading entity counts...
> 2019-06-11 09:55:28  Creating workers...
> 2019-06-11 09:55:28  Initializing global model...
> 2019-06-11 09:55:32  Starting epoch 1 / 50 edge path 1 / 1 edge chunk 1 / 1
> 2019-06-11 09:55:32  edge_path= edges_partitioned
> still in queue: 0
> 2019-06-11 09:55:32  Swapping partitioned embeddings None ( 0 , 0 )
> 2019-06-11 09:55:32  Loading entities

So , the training exited when the process start loading entites .
i'm pretty sure it's a memory problem . is it possible to batch the training and do not load all the entities at once ?
### Thank You !

**Comment by lerks** [comment]
You will notice that in the `fb15k_config.py` file the one and only entity type that is defined there, i.e., `all`, is configured with `'num_partitions': 1`. If you replace that `1` by a larger number then the `import_from_tsv` command will partition your edgelist in a larger number of smaller buckets.

Is this what you were looking for?

**Comment by laifi** [comment]
Thank you
